addons/mrp/migrations/10.0.2.0/pre-migration.py

```diff
@@ -36,16 +38,32 @@
     ('mrp.workorder', 'mrp_workorder', 'date_planned', 'date_planned_start'),
 ]
 
-_model_renames = [
-    ('mrp_production_workcenter_line', 'mrp_workorder'),
-]
-
 _xmlid_renames = [
     ('base.menu_mrp_config', 'mrp.menu_mrp_config'),
     ('base.menu_mrp_root', 'mrp.menu_mrp_root'),
 ]
 
 
+def rename_mrp_workorder(env):
+    """Special case where in v9, you have a model called mrp.workorder that is
+    an SQL view report, but on v10 it's a model, which corresponds to a
+    renamed model.
+    """
+    cr = env.cr
+    tools.drop_view_if_exists(cr, 'mrp_workorder')
+    openupgrade.rename_tables(
+        cr, [('mrp_production_workcenter_line', 'mrp_workorder')],
+    )
+    openupgrade.logged_query(
+        cr,
+        "DELETE FROM ir_model WHERE model = %s",
+        ('mrp.workorder', )
+    )
+    openupgrade.rename_models(
```

Comment:
This now errors with the following after pre-migration (module own upgrade, before post-migration, it seems):

```
2018-01-16 10:23:20,719 28239 INFO migrate9_migrated odoo.modules.loading: loading mrp/security/ir.model.access.csv
2018-01-16 10:23:22,495 28239 INFO migrate9_migrated odoo.sql_db: bad query: INSERT INTO "ir_model_access" ("id", "perm_read", "name", "active", "perm_unlink", "perm_write", "p
erm_create", "group_id", "create_uid", "write_uid", "create_date", "write_date") VALUES(nextval('ir_model_access_id_seq'), true, 'mrp.workorder.user', true, true, true, true, 3
5, 1, 1, (now() at time zone 'UTC'), (now() at time zone 'UTC')) RETURNING id
2018-01-16 10:23:22,537 28239 INFO migrate9_migrated odoo.sql_db: bad query: INSERT INTO "ir_model_access" ("id", "perm_read", "name", "active", "perm_unlink", "perm_write", "p
erm_create", "group_id", "create_uid", "write_uid", "create_date", "write_date") VALUES(nextval('ir_model_access_id_seq'), true, 'mrp.workorder', true, true, true, true, 36, 1,
 1, (now() at time zone 'UTC'), (now() at time zone 'UTC')) RETURNING id
2018-01-16 10:23:23,943 28239 ERROR migrate9_migrated odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/registry.py", line 82, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 409, in load_modules
    force, status, report, loaded_modules, update_module, upg_registry)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 299, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks, upg_registry=upg_registry)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 98, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/var/tmp/openupgrade/10.0/server/odoo/tools/convert.py", line 843, in convert_file
    convert_csv_import(cr, module, pathname, fp.read(), idref, mode, noupdate)
  File "/var/tmp/openupgrade/10.0/server/odoo/tools/convert.py", line 899, in convert_csv_import
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))
Exception: Module loading mrp failed: file mrp/security/ir.model.access.csv could not be processed:
 No matching record found for external id 'model_mrp_workorder' in field 'Object'
Missing required value for the field 'Object' (model_id)
No matching record found for external id 'model_mrp_workorder' in field 'Object'
Missing required value for the field 'Object' (model_id)
2018-01-16 10:23:23,951 28239 CRITICAL migrate9_migrated odoo.service.server: Failed to initialize database `migrate9_migrated`.
Traceback (most recent call last):
  File "/var/tmp/openupgrade/10.0/server/odoo/service/server.py", line 914, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/registry.py", line 82, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 409, in load_modules
    force, status, report, loaded_modules, update_module, upg_registry)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 299, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks, upg_registry=upg_registry)
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/var/tmp/openupgrade/10.0/server/odoo/modules/loading.py", line 98, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/var/tmp/openupgrade/10.0/server/odoo/tools/convert.py", line 843, in convert_file
    convert_csv_import(cr, module, pathname, fp.read(), idref, mode, noupdate)
  File "/var/tmp/openupgrade/10.0/server/odoo/tools/convert.py", line 899, in convert_csv_import
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))
Exception: Module loading mrp failed: file mrp/security/ir.model.access.csv could not be processed:
 No matching record found for external id 'model_mrp_workorder' in field 'Object'
Missing required value for the field 'Object' (model_id)
No matching record found for external id 'model_mrp_workorder' in field 'Object'
Missing required value for the field 'Object' (model_id)
2018-01-16 10:23:23,954 28239 INFO migrate9_migrated odoo.service.server: Initiating shutdown
2018-01-16 10:23:23,954 28239 INFO migrate9_migrated odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.
```

Reply:
have you found the way to overcome this?

Reply:
@leio Just fixed it here: https://github.com/OCA/OpenUpgrade/pull/1076/commits/7fe9dc234b78bbbc314421bf716516a44f26b9ee